docs/navigation-configuration.md

```diff
@@ -1,22 +1,239 @@
+These are the elements of the Luigi navigation: 
 
-These are the elements of the Luigi navigation:
 
-- A top navigation
-- A side navigation
-- A main content window
+- Top navigation, where the main navigation path is displayed.
+- Side navigation, where the application are defined.
+- Main content window, where the micro front-end is rendered.
 
-The image shows where you can use various methods and parameters to fill the navigation or display a micro front-end.
+The image shows where in the navigation you can use various methods and parameters to create the navigation structure or display a micro front-end.  
 
 ![Navigation layout](assets/navigation-structure.png)
 
-This code sample demonstrates your options when configuring navigation for Luigi.
+
+# Navigation structure
+
+The navigation structure is a recursive tree-like data structure that defines all possible navigation paths within the application.
+
+A navigation path is any existing path in the navigation tree. It connects the following elements together:
+
+- The **pathSegment** of the main application. The **pathSegment** is a part of the URL in the browser URL bar.
+- The **viewUrl** of a micro front-end rendered in the content area of the application.
+
+A sample navigation structure looks as follows:
+
+````
+{
+  navigation: {
+    nodes: [
+      {
+        pathSegment: 'home',
+        label: 'Home',
+        viewUrl: 'https://my.microfrontend.com/',
+        children: [
+          {
+            pathSegment: 'settings',
+            label: 'Settings',
+            viewUrl: 'https://my.microfrontend.com/general/settings.html'
+          },
+          {
+            pathSegment: 'projects',
+            label: 'Projects',
+            viewUrl: 'https://my.microfrontend.com/projects/list.html',
+            children: [
+              {
+                pathSegment: 'pr1',
+                label: 'Project one',
+                viewUrl: 'https://my.microfrontend.com/projects/details.html#id=pr1'
+              },
+              {
+                pathSegment: 'pr2',
+                label: 'Project two',
+                viewUrl: 'https://my.microfrontend.com/projects/details.html#id=pr2'
+              }
+            ]
+          }
+        ]
+      }
+    ]
+  }
+}
+````
+## Node navigation
+
+When you navigate between Nodes that are located on the same domain, Luigi triggers a hash or path change. Then it sends an updated context in order not to fully reload the view for a single-page application based micro front-end. Navigation between domains triggers a full page load in order to comply with cross-domain security concepts.
```

Comment:
Nodes should be lower case

Reply:
Done - all mentions of "node" are now lowercase.